File: src/main/org/nlogo/prim/gui/_exportinterface.java
package org.nlogo.prim.gui ;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.EngineException;
import org.nlogo.nvm.Syntax;
import org.nlogo.window.GUIWorkspace;

public final strictfp class _exportinterface
	extends org.nlogo.nvm.Command
{
	@Override
	public Syntax syntax()
	{
		int[] right = { Syntax.TYPE_STRING } ;
		return Syntax.commandSyntax( right ) ;
	}
	@Override
	public void perform( final org.nlogo.nvm.Context context )
		throws LogoException
	{
		if( ! ( workspace instanceof GUIWorkspace ) )
		{
			throw new EngineException(
				context , this , token().name() + " can only be used in the GUI" ) ;
		}
		( (GUIWorkspace) workspace).updateUI() ;
		final String filePath = argEvalString( context , 0 ) ;
		workspace.waitFor
			( new org.nlogo.api.CommandRunnable() {
					public void run() throws LogoException {
						try
						{
							workspace.exportInterface
								( workspace.fileManager().attachPrefix( filePath ) ) ;
						}
						catch( java.io.IOException ex )
						{
							throw new EngineException
								( context , _exportinterface.this ,
								  token().name() + 
								  ": " + ex.getMessage() ) ;
						}
					} } ) ;
		context.ip = next ;
	}
}

File: src/main/org/nlogo/prim/gui/_exportinterface.scala
package org.nlogo.prim.gui

import org.nlogo.api.CommandRunnable
import org.nlogo.nvm.{ Command, Context, EngineException }
import org.nlogo.nvm.Syntax
import org.nlogo.window.GUIWorkspace

class _exportinterface extends Command {

  override def syntax =
    Syntax.commandSyntax(Array(Syntax.TYPE_STRING))

  override def perform(context: Context) {
    workspace match {
      case gw: GUIWorkspace =>
        gw.updateUI()
        val filePath = argEvalString(context, 0)
        workspace.waitFor(
          new CommandRunnable() {
            override def run() {
              try workspace.exportInterface(
                workspace.fileManager.attachPrefix(filePath))
              catch {
                case e: java.io.IOException =>
                  throw new EngineException(
                    context, _exportinterface.this, token.name + ": " + e.getMessage)
              }}})
      case _ =>
        throw new EngineException(
          context, this, token.name + " can only be used in the GUI")
    }
    context.ip = next
  }
}
